contrib/win32/win32compat/shell-host.c

```diff
@@ -980,6 +1025,20 @@ ConsoleEventProc(HWINEVENTHOOK hWinEventHook,
 }
 
 DWORD 
+GetCurrentTerminalType(void)
+{
```

Comment:
> so if we take a dependency on the "TERM" then you need to modify the ssh client code to send the "TERM" as "xterm" when the "TERM" environment variable is not found

So you do not need to do this. In this way the most I just separates **Putty** (and others) from **OpneSSH-Win32**.